### PR TITLE
feat(onboarding): add issue worktree bootstrap helper

### LINKED ISSUE #1769: feat(onboarding): add issue-to-worktree bootstrap helper
## Description
Implement an LLM-friendly, independently shippable backlog item for **add issue-to-worktree bootstrap helper**. This issue should stay narrow enough for one agent/PR while producing durable behavior, tests, and documentation where appropriate.

## Affected locations
- docs/onboarding and contributor guides
- CLI/help output or onboarding scripts
- PM/worker handoff templates

## Impact
Without this work, future agents and operators will keep rediscovering the same gap, creating avoidable coordination cost, reliability risk, or ambiguous handoffs.

## Recommendation
Implement the scoped behavior with deterministic tests, concise docs, and structured output that PM/liveness tooling can consume.

## Goals
Deliver add issue-to-worktree bootstrap helper as a discrete capability or hardening change.
Make the behavior deterministic enough for autonomous workers to verify.
Keep the implementation small enough for a one-issue worker and one PR.
Capture operator/user-facing behavior in docs or help text when applicable.

## Expected outcomes
add issue-to-worktree bootstrap helper is available, documented, and test-backed.
Future PM/worker handoffs can point at this issue without needing extra interpretation.
Failure modes are explicit and produce actionable messages instead of silent drift.
The change does not broaden into adjacent backlog work or unrelated refactors.

## Suggested implementation slices
Inventory the current code/docs paths that own this behavior.
Add or update the narrow model/config/schema needed for the feature or guardrail.
Implement the smallest runtime/docs change that satisfies the acceptance criteria.
Add focused fixtures/tests for success, failure, and edge cases.
Update docs/help output and handoff notes so later agents can use it.

## Acceptance criteria
- [ ] There is a dedicated implementation or documentation change for add issue-to-worktree bootstrap helper.
- [ ] At least one automated test, fixture, or deterministic verifier covers the intended behavior.
- [ ] At least one negative/edge case is covered so the failure mode is explicit.
- [ ] Docs/help text or inline operator guidance explains how to use or interpret the behavior.
- [ ] The PR description links this issue and summarizes verification evidence.
- [ ] No unrelated production behavior is changed outside the stated scope.

## LLM handoff notes
- Treat this as a one-issue/one-PR unit.
- Start by searching for existing implementations and tests before adding new surfaces.
- Prefer structured outputs and fixtures over prose-only behavior.
- If this overlaps an existing open issue, comment with evidence and stop for PM deduplication instead of creating an unscoped implementation.
- Do not perform broad refactors or change unrelated worker/PM behavior while implementing this ticket.

## Source
Created by the 2026-07-11 fourth strategic backlog expansion run requesting 100 issues. Topic: **Onboarding**.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- Adds `scripts/issue-worktree-bootstrap.mjs` and `npm run issue:worktree` for deterministic one-issue worktree setup.
- Emits structured dry-run output with branch/worktree/duplicate-check/verification commands and rejects invalid issue/ref/repo inputs.
- Documents the helper in onboarding and the GitHub issue-fixing guide.

## Verification
- `npm run test:root -- tests/unit/issue-worktree-bootstrap.test.ts`
- `node scripts/issue-worktree-bootstrap.mjs --dry-run --json --issue 1769 --title 'feat(onboarding): add issue-to-worktree bootstrap helper'`
- `npm run lint` (passes; existing warnings only)
- `npm run typecheck`
- `npm run build`

Closes #1769

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: scripts/issue-worktree-bootstrap.mjs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    ['gh', 'pr', 'list', '--repo', repo, '--state', 'open', '--search', `${issue} in:body`, '--json', 'number,title,headRefName,url'],
+    ['git', 'branch', '--list', branch],
+  ];
+  const create = reuse
+    ? [['git', 'worktree', 'add', worktreePath, branch]]
+    : [['git', 'worktree', 'add', '-b', branch, worktreePath, base]];
+  const verify = [
+    ['git', '-C', worktreePath, 'status', '--short', '--branch'],
+    ['git', '-C', worktreePath, 'config', 'user.name', 'David Mendez'],
+    ['git', '-C', worktreePath, 'config', 'user.email', 'me@davidmendez.dev'],
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs]: Fixed in 17a90d30: the helper now enables worktree config and writes identity with `git config --worktree`, so user.name/user.email are scoped to the created linked worktree rather than the shared repository config.

File: scripts/issue-worktree-bootstrap.mjs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    if (existsSync(plan.worktreePath) && !plan.reuse) {
+      throw new Error(`Refusing to overwrite existing worktree path: ${plan.worktreePath}. Use --reuse only for an existing branch.`);
+    }
+    mkdirSync(resolve(plan.worktreePath, '..'), { recursive: true });
+
+    for (const command of plan.commands.preflight) runCommand(command, process.cwd());
+    assertNoDuplicateIssueWork(plan);
+    for (const command of plan.commands.create) runCommand(command, process.cwd());
+    for (const command of plan.commands.verify) runCommand(command, process.cwd());
+    console.log(`[issue-worktree] ready: ${plan.worktreePath}`);
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs]: Fixed in 17a90d30: `--json` real runs now emit a single JSON object on stdout after success, while human status output is only used for non-JSON mode. Dry-run JSON remains JSON-only.

File: scripts/issue-worktree-bootstrap.mjs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    throw new Error(`Repository must be OWNER/REPO, got: ${repo}`);
+  }
+
+  const preflight = [
+    ['git', 'rev-parse', '--is-inside-work-tree'],
+    ['git', 'fetch', remote, base.replace(new RegExp(`^${remote}/`, 'u'), '')],
+  ];
+  const duplicateChecks = [
+    ['gh', 'pr', 'list', '--repo', repo, '--state', 'open', '--search', `${issue} in:body`, '--json', 'number,title,headRefName,url'],
+    ['git', 'branch', '--list', branch],
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs]: Fixed in 17a90d30: duplicate branch detection now runs `git branch --all --list` and checks both the local branch and `remotes/<remote>/<branch>` before creating a new branch.

File: scripts/issue-worktree-bootstrap.mjs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  }
+  return issue;
+}
+
+export function buildIssueWorktreePlan(options) {
+  const issue = parseIssueNumber(options.issue);
+  const titleSlug = slugifyTitle(options.title ?? `issue-${issue}`);
+  const branch = options.branch ?? `resolve/issue-${issue}-${titleSlug}`;
+  const remote = options.remote ?? DEFAULT_REMOTE;
+  const base = options.base ?? DEFAULT_BASE;
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs]: Fixed in 17a90d30: the default base now derives from the selected remote as `<remote>/main`; for example `--remote upstream` plans and fetches `upstream/main`.

File: scripts/issue-worktree-bootstrap.mjs [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  if (!Number.isSafeInteger(issue) || issue <= 0) {
+    throw new Error('Issue number must be a positive integer.');
+  }
+  return issue;
+}
+
+export function buildIssueWorktreePlan(options) {
+  const issue = parseIssueNumber(options.issue);
+  const titleSlug = slugifyTitle(options.title ?? `issue-${issue}`);
+  const branch = options.branch ?? `resolve/issue-${issue}-${titleSlug}`;
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs:39]: Fixed in 17a90d30: generated title slugs are capped before branch construction, with regression coverage for long issue titles.

**Comment by djm204** [comment]
@codex review

File: scripts/issue-worktree-bootstrap.mjs [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+  if (!/^[-./A-Za-z0-9_]+$/u.test(branch) || branch.includes('..') || branch.startsWith('-') || branch.endsWith('/')) {
+    throw new Error(`Unsafe branch name for issue #${issue}: ${branch}`);
+  }
+  if (!/^[A-Za-z0-9_.-]+\/[A-Za-z0-9_.-]+$/u.test(repo)) {
+    throw new Error(`Repository must be OWNER/REPO, got: ${repo}`);
+  }
+
+  const preflight = [
+    ['git', 'rev-parse', '--is-inside-work-tree'],
+    ['git', 'fetch', remote, base.replace(new RegExp(`^${remote}/`, 'u'), '')],
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs:62]: Fixed in 710edee10b21ecf076521ed99fe9fb63aa96ef6c: the helper now fetches all remote heads before duplicate/reuse checks, so an existing remote issue branch is visible before branch creation.

File: scripts/issue-worktree-bootstrap.mjs [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+
+function assertNoDuplicateIssueWork(plan) {
+  const [openPrCommand, branchCommand] = plan.commands.duplicateChecks;
+  const openPrOutput = runCommand(openPrCommand, process.cwd());
+  const openPrs = JSON.parse(openPrOutput || '[]');
+  if (Array.isArray(openPrs) && openPrs.length > 0) {
+    const summary = openPrs
+      .map((pr) => `#${pr.number ?? '?'} ${pr.headRefName ?? ''} ${pr.url ?? ''}`.trim())
+      .join('; ');
+    throw new Error(`Issue #${plan.issue} already appears in an open PR: ${summary}`);
```
[the remark this answers is not in the document]
Comment (djm204) [reply scripts/issue-worktree-bootstrap.mjs]: Fixed in 710edee10b21ecf076521ed99fe9fb63aa96ef6c: reuse mode now allows an open PR when its headRefName matches the requested branch, while still rejecting PRs for the same issue on any different branch. Added unit coverage for both cases.

**Comment by djm204** [comment]
@codex review
